interview-questions.md

```diff
@@ -7,90 +7,112 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: 
+    super() is used if you're creating a new class that you want to inherit an existing class's constructors.
 
-  Researched answer:
+  Researched answer: 
+    The method super() is used when extending a parent class into a new child class you want to create. Using super() in the child class will inherit the parent class' constructors. super() can also be used to call on methods in parent classes within child classes.
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
+  Your answer: 
+    DOM is Document Object Model. Not sure what virtual DOM specifically refers to.
 
   Researched answer:
+    The DOM is an object-oriented representation of your page, that can be modified with a scripting language like JavaScript. A virtual DOM refers to an "ideal" version of your page. This is synced with the actual DOM for your page with something like ReactDOM. This might be wrong, still a little shakey on DOM.
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: 
+    The only two higher-order functions that come to mind are .map() and .filter which return new arrays after manipulating the data. 
+  
+  Researched answer: 
+    A higher-order function is any function that operates on other functions, either by taking them as an argument or returning them. .map(), .filter(), and .reduce() are all higher order because they can take other functions as their argument.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: 
+    JSX is basically html written within a React app and rendered onto an HTML page through a single container. Used so that JavaScript can better interact with HTML elements and elements of the page can rerender without the entire page reloading.
 
   Researched answer:
+    JSX is an syntax extension for JavaScript that allows you to render html tags on a page as React elements. It is not necessary to use React but is widely used as a visual aid for UI when working on JavaScript.
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: 
+    React is a JavaScript library that lets you create web applications.
```

Comment:
Totally. I'd want to expand upon this a bit more, as its specifically for creating *react*ive front-end applications.